landlab/components/dimensionless_discharge/dimensionless_discharge.py

```diff
@@ -0,0 +1,196 @@
+     empirically-derived by comparing dimensionless discharge estimates
+     against previous debris flow events.  The values will vary based on
+     local geology and soil type. In southern California values of C=12
+     and N=0.85 (upper limits, Tang et al, 2019) and C=4.29, N=0.78
+     (lower limits, Tang et al, 2019) have been used, while values of
+     C=0.195 and N=1.27 have been in used in the Italian Dolomites
+     (Gregoretti and Fontana, 2008). Default values are C=12 and N=0.85.
+
+     Parameters
+     ----------
+     soil_density : float list
+         Density of soil in watershed (kg/m^3)
+     water_density : float
+         density of water in waterhed (kg/m^3)
+     C : float
+         Numerator of the debris flow threshold equation; Empirically
+         derived constant (See Tang et al. 2019)
+     N : float
+         Exponent for slope in the denominator of the debris flow
+         threshold equation; Empirically derived constant (See Tang et
+         al. 2019)
+
+     Examples
+     --------
+     >>> from landlab.components import DimensionlessDischarge
+     >>> from landlab import RasterModelGrid
+     >>> import random
+     >>> watershed_grid = RasterModelGrid((3, 3))
+     >>> flux = watershed_grid.add_ones('node', 'flux')
+     >>> d50 = watershed_grid.add_ones('node', 'd50')
+     >>> watershed_grid.at_node ['dem_values'] = \
+     ... np.array([[1.1, 2, 3, 4, 2, 3, 4, 5, 3]])
+     >>> dd = DimensionlessDischarge(watershed_grid)
+     >>> dd.run_one_step()
+     >>> print(watershed_grid.at_node['dimensionless_discharge'])
+     [ 0.55372743  0.55372743  0.55372743  0.55372743  0.55372743
+             0.55372743  0.55372743  0.55372743  0.55372743]
+
+
+
+     References
+     ----------
+     Tang, H., McGuire, L. A., Rengers, F. K., Kean, J. W., Staley,
+     D. M., & Smith, J. B. (2019). Developing and Testing Physically
+     Based Triggering Thresholds for Runoff-Generated Debris Flows.
+     Geophysical Research Letters, 46(15), 8830–8839.
+     https://doi.org/10.1029/2019GL083623
+
+    """
+
+    _name = "DimensionlessDischargeModel"
+
+    _unit_agnostic = False
+
+    _info = {
+        "dimensionless_discharge": {
+            "dtype": float,
+            "intent": "out",
+            "optional": False,
+            "units": "none",
+            "mapping": "node",
+            "doc": "Dimensionless discharge value for a stream segment.",
+        },
+        "dimensionless_discharge_above_threshold": {
+            "dtype": bool,
+            "intent": "out",
+            "optional": False,
+            "units": "none",
+            "mapping": "node",
+            "doc": "Dimensionless discharge value for a stream segment.",
+        },
+        "flux": {
+            "dtype": float,
+            "intent": "in",
+            "optional": False,
+            "units": "m/s",
+            "mapping": "node",
+            "doc": "flux in each stream segment",
+        },
+        "d50": {
```

Comment:
Same as above. This needs a more descriptive name.

Reply:
maybe "grain_size__median"... we should use something decent for this standard name since it is a standard value. What do you think @mcflugen

Reply:
@kbarnhart We should. Maybe *sediment_grain__d50_diameter*? It's an existing name.

Reply:
If that is an existing name, then yes, that's what should be used. Is that a CSDMS standard name?... I couldn't find a component that used it?

Reply:
Yes, a CSDMS Standard Name (the existing name is actually, *channel_bottom_sediment_grain__d50_diameter*). There aren't any *landlab* components that use it.

Reply:
@eculler Thoughts on this naming too?  channel_bottom_sediment_grain__d50_diameter sounds accurate to me.